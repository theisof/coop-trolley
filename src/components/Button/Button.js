import React from 'react'

const Button = ({
  children,
  style = {},
  type = 'primary',
  className = '',
  onClick = () => {}
}) => (
  <div
    className={`coop-button coop-button--${type} ${className}`}
    style={style}
    onClick={onClick}
  >
    { children }
  </div>
)

export default Button


