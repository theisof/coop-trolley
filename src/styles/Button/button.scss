.coop-button {
  font-family: $font-standard;
  background-color: $color-primary;
  width: 100%;
  padding: 10px;
  color: #fff;
  border-radius: 5px;
  text-align: center;
  font-size: 16px;
  cursor: pointer;
  font-weight: bold;
  transition: background-color 0.5s, border-color 0.5s;
  border-width: 2px;
  border-color: $color-primary;
  border-style: solid;
  display: flex;
  justify-content: center;
  text-decoration: none;
  box-sizing: border-box;

  &:focus,
  &:active,
  &:hover {
    opacity: 0.5;
    text-decoration: none;
    color: #fff;
  }
}

.coop-button--secondary {
  background-color: #fff;
  color: $color-primary;

  &:focus,
  &:active,
  &:hover {
    color: $color-primary;
  }
}

.coop-button--white-transparent {
  background-color: transparent;
  color: #fff;
  border-color: #fff;
}
