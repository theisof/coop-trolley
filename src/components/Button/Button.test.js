import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json';
import Button from './Button'

it('Button renders correctly', () => {
  const tree = shallow( <Button />)

  expect(toJSON(tree)).toMatchSnapshot()
})
