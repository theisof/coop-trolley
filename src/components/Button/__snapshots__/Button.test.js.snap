// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div
  className="coop-button coop-button--primary "
  onClick={[Function]}
  style={Object {}}
/>
`;
